stub-idp/src/main/java/uk/gov/ida/stub/idp/repositories/AllIdpsUserRepository.java

```diff
@@ -112,6 +112,17 @@ public DatabaseEidasUser createUserForStubCountry(String countryFriendlyName,
         return Optional.empty();
     }
 
+    public Optional<DatabaseIdpUser> getUserForIdpByPersistentId(String idpFriendlyName, String persistentId) {
+        final List<DatabaseIdpUser> matchingUsers = userRepository.getUsersForIdp(idpFriendlyName)
+                .stream()
+                .filter(u -> u.getPersistentId().equalsIgnoreCase(persistentId))
+                .collect(Collectors.toList());
+        if (!matchingUsers.isEmpty()) {
```

Comment:
you can .findFirst() at the end of the lambda instead of collect() and just return that.